test/specs/basic/ignore-joiners-synonyms.js

```diff
@@ -0,0 +1,46 @@
+/*!***************************************************
+ * mark.js
+ * https://github.com/julmot/mark.js
+ * Copyright (c) 2014–2016, Julian Motz
+ * Released under the MIT license https://git.io/vwTVl
+ *****************************************************/
+"use strict";
+describe("basic mark with ignore joiners synonyms", function () {
+    var $ctx1, $ctx2;
+    beforeEach(function (done) {
+        loadFixtures("basic/ignore-joiners-synonyms.html");
+
+        $ctx1 = $(".basic-ignore-joiners-synonyms > div:nth-child(1)");
+        $ctx2 = $(".basic-ignore-joiners-synonyms > div:nth-child(2)");
+        new Mark($ctx1[0]).mark("Lorem", {
+            "synonyms": {
+                "Lorem": "ipsum"
+            },
+            "separateWordSearch": false,
+            "diacritics": false,
+            "ignoreJoiners": true,
+            "caseSensitive": true,
```

Comment:
Is it true, that you're mixing `"caseSensitive"` with `"synonyms"` here? What's the reason?